docs/anti-patterns.mdx

```diff
@@ -4,35 +4,110 @@ title: Cadence Anti-Patterns
 
 This is an opinionated list of issues that can be improved if they are found in Cadence code intended for production.
 
-## Security and Robustness
+# Security and Robustness
 
-### Events From Resources Might Not Be Unique
+## Avoid using `AuthAccount` as a function parameter
 
-A public function on a contract that creates a resource can be called by any account.
+### Problem
 
-If that resource has functions that emit events, any account can create an instance of that resource and emit those events.
+Some projects want to authenticate users using account addresses or want to perform operations for users directly.
+A not-uncommon reaction is to expect a user to pass their `AuthAccount` object as a function parameter
+to a contract function so that address can be queried or so an object can be stored directly.
+This is a problem, because the `AuthAccount` object can be used to access ANY private area of an account,
+meaning that a project could do something with your `AuthAccount` that you don't intend it to do.
+
+### Solution
+
+Projects should find other ways to authenticate users, such as using resources and capabilities as authentication objects.
+They should also expect to perform most storage and linking operations within transaction bodies
+instead of in a contract utility functions.
+
+There are some scenarios where using an `AuthAccount` object is necessary, such as a cold storage multi-sig,
+but those cases are extremely rare and `AuthAccount` usage should still be avoided unless absolutely necessary.
```

Comment:
Other solutions are available by utilizing resources and capabilities instead of `AuthAccount` for authentication purposes. Most storage and linking operations can be limited to transactions instead of calling upon contract utility functions. 

There are some scenarios where using an `AuthAccount` object is necessary such as a cold storage multi-sig. Again, there are rare cases where that's applicable, and it's recommended not to use `AuthAccount` unless absolutely necessary.